subprojects/gradle/runner/client/src/main/kotlin/com/avito/runner/scheduler/listener/ArtifactsTestListener.kt

```diff
@@ -43,17 +45,24 @@ internal class ArtifactsTestListener(
         result: TestCaseRun.Result,
         durationMilliseconds: Long,
         executionNumber: Int,
-        testMetadataDirectory: File,
-        testFolder: String
+        testArtifactsDir: Result<File>
     ) {
+        val tempDirectory = createTempDirectory()
+
         val testResult = when (result) {
             Passed,
-            is Failed.InRun ->
-                pullArtifacts(
-                    device = device,
-                    runnerOutputDirectory = testMetadataDirectory,
-                    testFolder = testFolder
-                )
+            is Failed.InRun -> {
+                val artifacts = testArtifactsDir.flatMap { dir ->
+
+                    // last /. means to adb to copy recursively, and not to copy last dir
+                    // example:
+                    //  - from: /sdcard/Android/someDir/ to: /xx ; will copy to /xx/someDir/ and not recursive
+                    //  - from: /sdcard/android/someDir/. to: /xx ; will copy to /xx and recursive
```

Comment:
Is it ADB specific? 
If so, let's keep this knowledge in `Device.pull` kdoc but not in clients.